WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationMLSMessageAddEventProcessor.swift

```diff
@@ -32,9 +34,133 @@ protocol ConversationMLSMessageAddEventProcessorProtocol {
 
 struct ConversationMLSMessageAddEventProcessor: ConversationMLSMessageAddEventProcessorProtocol {
 
-    func processEvent(_: ConversationMLSMessageAddEvent) async throws {
-        // TODO: [WPB-10172]
-        assertionFailure("not implemented yet")
+    let conversationLocalStore: any ConversationLocalStoreProtocol
+    let messageLocalStore: any MessageLocalStoreProtocol
+    let userLocalStore: any UserLocalStoreProtocol
+    let protobufMessageProcessor: any ConversationProtobufMessageProcessorProtocol
+
+    func processEvent(_ event: ConversationMLSMessageAddEvent) async throws {
+        let conversationID = event.conversationID
+        let senderID = event.senderID
+        let date = event.timestamp
+        let decryptedMessages = event.decryptedMessages
+        
+        guard !decryptedMessages.isEmpty else {
+            return WireLogger.proteus.warn(
+                "failed to add MLS message: there are no decrypted messages to process"
+            )
+        }
+
+        for decryptedMessage in decryptedMessages {
+            await processDecryptedMessage(
+                decryptedMessage,
+                conversationID: conversationID,
+                senderID: senderID,
+                date: date
+            )
+        }
+    }
+
+    private func processDecryptedMessage(
+        _ decryptedMessage: ConversationMLSMessageAddEvent.DecryptedMessage,
+        conversationID: ConversationID,
+        senderID: UserID,
+        date: Date?
+    ) async {
+        guard let conversation = await conversationLocalStore.fetchConversation(
+            id: conversationID.uuid,
+            domain: conversationID.domain
+        ) else {
+            return WireLogger.proteus.error(
+                "failed to add MLS message: conversation not found in db"
+            )
```

Comment:
I'm wondering if thIs should actually throw an error. If there's no conversation, then it seems like a pretty serious error.

Reply:
yes it makes sense, fixed in https://github.com/wireapp/wire-ios/pull/2164/commits/90da7addc7e950bb179e65ed21fe6583b88d9f4c